### PR TITLE
Upgrade to kotlin version 1.3.50

### PR DESCRIPTION
**Pull request by raviagarwal7**
- Upgrades to kotlin 1.3.50
- kotlin_home and kotlin_plugin_home are two different targets now.
- Kotlin all open plugin is disabled for now since it fails with the below error
```
BUILT 1219/1586 JOBS 13.5s //.okbuck/ext/org/robolectric:android-all-8.1.0-robolectric-4611349.jar__downloaded
DOWNLOADED 0 ARTIFACTS, 0.00 BYTES, 100.0% CACHE MISS
BUILDING: FINISHED IN 17.0s (100%) 1219/1586 JOBS, 1212 UPDATED
BUILD FAILED
Command failed with exit code 2.
stderr: warning: flag is not supported by this version of the compiler: -Xadd-compiler-builtins
warning: flag is not supported by this version of the compiler: -Xload-builtins-from-dependencies
logging: using Kotlin home directory /home/circleci/okbuck/buck-out/gen/.okbuck/workspace/kotlin_home/kotlin_home/libexec
logging: configuring the compilation environment
exception: java.lang.IllegalStateException: The provided plugin org.jetbrains.kotlin.allopen.AllOpenComponentRegistrar is not compatible with this version of compiler
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.registerExtensionsFromPlugins$cli(KotlinCoreEnvironment.kt:639)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$ProjectEnvironment.registerExtensionsFromPlugins(KotlinCoreEnvironment.kt:145)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.<init>(KotlinCoreEnvironment.kt:185)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.<init>(KotlinCoreEnvironment.kt:125)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.createForProduction(KotlinCoreEnvironment.kt:432)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.createCoreEnvironment(K2JVMCompiler.kt:253)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:151)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:54)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:84)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:104)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:82)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.facebook.buck.jvm.kotlin.JarBackedReflectedKotlinc.buildWithClasspath(JarBackedReflectedKotlinc.java:176)
	at com.facebook.buck.jvm.kotlin.KotlincStep.execute(KotlincStep.java:92)
	at com.facebook.buck.step.StepRunner.runStep(StepRunner.java:57)
	at com.facebook.buck.core.build.engine.impl.CachingBuildRuleBuilder$BuildRuleSteps.executeCommands(CachingBuildRuleBuilder.java:1448)
	at com.facebook.b
```

### CODE REVIEW

**Comment by raviagarwal7** [comment]
building the kotlin library fails with this error. @tyvsmith -- any ideas?

```
BUILT 1219/1586 JOBS 13.5s //.okbuck/ext/org/robolectric:android-all-8.1.0-robolectric-4611349.jar__downloaded
DOWNLOADED 0 ARTIFACTS, 0.00 BYTES, 100.0% CACHE MISS
BUILDING: FINISHED IN 17.0s (100%) 1219/1586 JOBS, 1212 UPDATED
BUILD FAILED
Command failed with exit code 2.
stderr: warning: flag is not supported by this version of the compiler: -Xadd-compiler-builtins
warning: flag is not supported by this version of the compiler: -Xload-builtins-from-dependencies
logging: using Kotlin home directory /home/circleci/okbuck/buck-out/gen/.okbuck/workspace/kotlin_home/kotlin_home/libexec
logging: configuring the compilation environment
exception: java.lang.IllegalStateException: The provided plugin org.jetbrains.kotlin.allopen.AllOpenComponentRegistrar is not compatible with this version of compiler
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.registerExtensionsFromPlugins$cli(KotlinCoreEnvironment.kt:639)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$ProjectEnvironment.registerExtensionsFromPlugins(KotlinCoreEnvironment.kt:145)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.<init>(KotlinCoreEnvironment.kt:185)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.<init>(KotlinCoreEnvironment.kt:125)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.createForProduction(KotlinCoreEnvironment.kt:432)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.createCoreEnvironment(K2JVMCompiler.kt:253)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:151)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:54)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:84)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:104)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:82)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.facebook.buck.jvm.kotlin.JarBackedReflectedKotlinc.buildWithClasspath(JarBackedReflectedKotlinc.java:176)
	at com.facebook.buck.jvm.kotlin.KotlincStep.execute(KotlincStep.java:92)
	at com.facebook.buck.step.StepRunner.runStep(StepRunner.java:57)
	at com.facebook.buck.core.build.engine.impl.CachingBuildRuleBuilder$BuildRuleSteps.executeCommands(CachingBuildRuleBuilder.java:1448)
	at com.facebook.buck.core.build.engine.impl.CachingBuildRuleBuilder$BuildRuleSteps.runWithDefaultExecutor(CachingBuildRuleBuilder.java:1425)
```

**Comment by tyvsmith** [comment]
That's the error that's produced when you're using a embeddable version of compiler plugins in the non embeddable version of the compiler (or vice versa)

File: libraries/kotlinlibrary/build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apply plugin: "kotlin"
 apply plugin: "kotlin-kapt"
-apply plugin: "kotlin-allopen"
+//apply plugin: "kotlin-allopen"
```
Comment (raviagarwal7) [inline libraries/kotlinlibrary/build.gradle]: Disabling kotlin-allopen plugin which doesn't seem to work

**Comment by raviagarwal7** [comment]
> That's the error that's produced when you're using an embeddable version of compiler plugins in the non embeddable version of the compiler (or vice versa)

Thanks, this worked. Creating plugins jars in a different target but still see similar issue.

**Comment by kageiit** [comment]
Lets hold off on this till https://github.com/facebook/buck/pull/2306#issuecomment-529931227 is resolved. the buck changes for single pass kapt have been reverted

**Comment by raviagarwal7** [comment]
Is being handled by https://github.com/uber/okbuck/pull/905

**Pull request closed without merging**